route53/main.tf

```diff
@@ -0,0 +1,45 @@
+resource "aws_route53_zone" "hosted_zone" {
+  count = var.create_hosted_zone == true ? 1 : 0
+  name  = var.hosted_zone_name
```

Comment:
Do we need a `hosted_zone_name` and `create_hosted_zone` variables. The only thing I'm thinking is that it's possible to have `create_hosted_zone` true and  `hosted_zone_name` which is going to cause some interesting behaviour.

Could you derive the `create_hosted_zone` from `hosted_zone_name` Something like
```hcl
locals {
  create_hosted_zone = var.hosted_zone_name != ""
}
```
and then use this instead of the variable.

I'm not 100% sure of this. One the one hand, it makes it hard to get in an invalid state but it is more "magic" behind the scenes. What do you think?

Reply:
@MancunianSam Not sure that is going to work, because it leads to a contradiction between these two resources:

```
resource "aws_route53_zone" "hosted_zone" {
  count = local.create_hosted_zone == true ? 1 : 0
  name  = var.hosted_zone_name

  tags = merge(
    var.common_tags,
    tomap(
      { "Name" = var.hosted_zone_name }
    )
  )
}

data "aws_route53_zone" "hosted_zone" {
  count = var.create_hosted_zone == true ? 0 : 1
  name  = var.hosted_zone_name
}
```
If the hosted zone already existing need the data resource, but it won't work without the hosted zone name variable, which is empty because its not being created.

Reply:
Yeah, fair point. Ok, ignore me.